src/EXTRA-FIX/fix_wall_flow.cpp

```diff
@@ -0,0 +1,294 @@
+  Fix(lmp, narg, arg),
+  flowax(FlowAxis::AX_X),
+  flowvel(0.0),
+  flowdir(0),
+  rndseed(0),
+  current_segment(nullptr)
+{
+  if (lmp->citeme) lmp->citeme->add(cite_fix_wall_flow_c);
+  if (narg < 9) utils::missing_cmd_args(FLERR, "fix wall/flow", error);
+
+  dynamic_group_allow = 1;
+  bool do_abort = false;
+
+  int iarg = 3;
+  // parsing axis
+  if (strcmp(arg[iarg], "x") == 0) flowax = FlowAxis::AX_X;
+  else if (strcmp(arg[iarg],"y") == 0) flowax = FlowAxis::AX_Y;
+  else if (strcmp(arg[iarg],"z") == 0) flowax = FlowAxis::AX_Z;
+  else error->all(FLERR,"Illegal fix wall/flow argument: axis must by x or y or z, but {} specified", arg[iarg]);
+
+  ++iarg;
+  // parsing velocity
+  flowvel = utils::numeric(FLERR,arg[iarg],do_abort,lmp);
+  if (flowvel == 0.0) error->all(FLERR,"Illegal fix wall/flow argument: velocity cannot be 0");
+  if (flowvel > 0.0) flowdir = 1;
+  else flowdir = -1;
+  if(flowdir < 0) error->all(FLERR, "Illegal fix wall/flow argument: negative direction is not supported yet");
+
+  ++iarg;
+  // parsing temperature
+  double flowtemp = utils::numeric(FLERR,arg[iarg],do_abort,lmp);
+  kT = lmp->force->boltz * flowtemp / force->mvv2e;
+
+  ++iarg;
+  // parsing seed
+  rndseed = utils::inumeric(FLERR, arg[iarg],do_abort,lmp);
+  if(rndseed <= 0) error->all(FLERR, "Illegal fix wall/flow argument: random seed must be positive integer");
+
+  ++iarg;
+  // parsing wall count
+  int wallcount = utils::inumeric(FLERR,arg[iarg],do_abort,lmp);
+  if(wallcount <= 0) error->all(FLERR,"Illegal fix wall/flow argument: wall count must be positive integer");
+
+  ++iarg;
+  // parsing walls
+  if(narg - iarg != wallcount && narg - iarg != wallcount + 2) error->all(FLERR, "Wrong fix wall/flow wall count");
+  auto getscale = [&]() -> double {
+      switch (flowax)
+      {
+        case FlowAxis::AX_X:
+          return domain->lattice->xlattice;
+        case FlowAxis::AX_Y:
+          return domain->lattice->ylattice;
+        case FlowAxis::AX_Z:
+          return domain->lattice->zlattice;
+        default: return 0.0;
+      }
+      return 0.0;
+  };
+  double scale = getscale();
+
+  if (narg - iarg == wallcount + 2)
+  {
+    if(strcmp(arg[narg - 2], "units") != 0) error->all(FLERR, "Wrong fix wall/flow units command");
+    if (strcmp(arg[narg - 1], "box") == 0) scale = 1.0;
+    else if (strcmp(arg[narg - 1], "lattice") == 0)
+    {
+      scale = getscale();
+    }
+    else error->all(FLERR, "Wrong fix wall/flow units command");
+  }
+
+  walls.resize(wallcount + 2);
+  walls.front() = domain->boxlo[flowax];
+  for (size_t w = 1; w <= wallcount; ++w, ++iarg)
+  {
+    walls[w] = utils::numeric(FLERR,arg[iarg],do_abort,lmp) * scale;
+  }
+  walls.back() = domain->boxhi[flowax];
+  if (!std::is_sorted(walls.begin(), walls.end(), std::less_equal<double>()))
+  {
+    error->all(FLERR, "Wrong fix wall/flow wall ordering or some walls are outside simulation domain");
+  }
+  std::cout << "Walls:\n";
```

Comment:
This has two problems:
- output should only be created by MPI rank 0, i.e. for `comm->me == 0` so that it is not replicated for parallel runs
- informational output should use the `utils::logmesg()` function
- i/o stream formatting is clumsy and slow.